homeassistant/components/imap/coordinator.py

```diff
@@ -373,7 +373,7 @@ async def _async_wait_push_loop(self) -> None:
             except InvalidFolder as ex:
                 _LOGGER.warning("Selected mailbox folder is invalid")
                 await self._cleanup()
-                self.config_entry.async_set_state(
+                self.config_entry._async_set_state(  # pylint: disable=protected-access
```

Comment:
Can this be fix in a follow-up to not manually set the state?

Reply:
Oh, that's already discussed in <https://github.com/home-assistant/core/pull/96724>